week3/hackyourweather/src/components/input/Input.jsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import Button from '../button/Button';
+import CardList from '../card-list/CardList';
+
+import './Input.css';
+
+const Input = () => {
+  const [city, setCity] = useState(``);
+  const [weatherData, setWeatherData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState({ error: false });
+
+  const handleError = (message) => setIsError({ error: true, message });
+
+  const removeCity = (id, data) => {
+    const newCityList = weatherData.filter((city) => city.id !== id);
+
+    setWeatherData([...newCityList]);
+  };
+
+  const showWeather = async (ev) => {
+    ev.preventDefault();
+
+    const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${process.env.REACT_APP_OPENWEATHERMAP_API_KEY}
+`;
+
+    try {
+      setIsLoading(true);
+
+      const resp = await fetch(url);
```

Comment:
I will leave you with no further request changes, as it looks good to me, one final thought that I think it's worth mentioning and I didn't notice before.

try to input a mistakenly city name, you will have an error, yep that's true, but if you try and enter the name as is, you will have the error again!

to fix this you could try something here:
```suggestion
    try {
      setIsLoading(true);
      if (isError.error) setIsError({error: false});

      const resp = await fetch(url);
```

now of course you will check if there is any errors after the request and you only expect one type of error in this program, so this in this condition works well.